src/main/java/org/edgexfoundry/messaging/ZeroMQEventSubscriber.java

```diff
@@ -132,29 +143,32 @@ public void setZeromqAddressPort(String zeromqAddressPort) {
 		return subscriber;
 	}
 
-	private Event toEvent(String eventString) throws IOException, ClassNotFoundException {
-		byte[] data = Base64.getDecoder().decode(eventString);
-		ObjectInputStream in = new ObjectInputStream(new ByteArrayInputStream(data));
-		Event event = (Event) in.readObject();
-		in.close();
-		return event;
+	private void processJsonEvent(JsonNode node) throws IOException {
+		logger.info("JSON event received");
+		Event event = toEvent(node.get("Payload").binaryValue());
+		logger.debug("Event: " + event);
+		executeOnEvent(event);
+	}
+	
+	private void processEvent(JsonNode node) throws IOException {
+		logger.info("Event received");
+		Event event = mapper.convertValue(node, Event.class);
```

Comment:
This will get called when message is sent from export-distro because it doesn't wrap the message back into the MessageEnvelope. The message can still be json or cbor encoded. Will this handle either?

Should export distro wrap it back into the Message Envelope for consistency?

Reply:
I think the intent of our work was to change the rules engine and not the other services and change the rules engine to handle wrapped or unwrapped message whether CBOR or JSON.  So I added some extra logic to handle CBOR unwrapped coming through export distro.